habitat-baselines/habitat_baselines/rl/ppo/ppo_trainer.py

```diff
@@ -317,69 +231,44 @@ def _init_train(self, resume_state=None):
         ):
             os.makedirs(self.config.habitat_baselines.checkpoint_folder)
 
-        self._setup_actor_critic_agent(ppo_cfg)
-        if resume_state is not None:
-            self.agent.load_state_dict(resume_state["state_dict"])
-            self.agent.optimizer.load_state_dict(resume_state["optim_state"])
-        if self._is_distributed:
-            self.agent.init_distributed(find_unused_params=False)  # type: ignore
+        logger.add_filehandler(self.config.habitat_baselines.log_file)
```

Comment:
Why is add_filehandler added here and not at the entry point of training, much earlier in the code ?

Reply:
It is actually occurring at the same point in the code, the previous code in `self._setup_actor_critic_agent` is now happening in `self._create_agent`. I think this is the natural place in the code to manage aspects of the config such as creating directories and log files it requires.

Reply:
Ok, fair. The directories might not exist before then